### PR TITLE
Bdex/54161: Migrate eVSS calls to rated disabilities functionality (vets-api /rated_disabilities)

### PR DESCRIPTION
**Pull request by sethdarragile6**
## Summary

- This PR migrates eVSS calls to rated disabilities functionality to Lighthouse (or any provider!)
- Created by: Benefit Delivery Experience team
- Flipper switch 'disability_compensation_lighthouse_rated_disabilities_provider' should be enabled when LH migration goes live, then removed once functionality is verified

## Related issue(s)
-  [54161](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/54161)

## Testing done

- Unit tests for factory, provider and service created. Also manually tested via Postman and using the local instance of vets-website

## What areas of the site does it impact?
Disability claims

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by christinec-fftc — APPROVED** [review]

**Comment by christinec-fftc** [comment]
The team had a pairing session where we walked through this pr together. lgtm!

File: spec/lib/disability_compensation/providers/lighthouse_rated_disabilities_provider_spec.rb [hunk 1]
```diff
@@ -7,11 +7,5 @@
...
   let(:current_user) { build(:user, :loa3) }
 
-  # TODO: update when the LighthouseRatedDisabilitiesProvider is implemented
-  it 'behaves like a RatedDisabilitiesProvider' do
-    # it_behaves_like :rated_disabilities_provider
-    expect do
-      LighthouseRatedDisabilitiesProvider.new(current_user)
-    end.to raise_error NotImplementedError
-  end
+  it_behaves_like 'rated disabilities provider'
```
Comment (mchae-nava) [inline spec/lib/disability_compensation/providers/lighthouse_rated_disabilities_provider_spec.rb:10]: Is `it_behaves_like` effective on its own? If I remember, it shares examples with the current context, but then it requires `shared_examples` and the relevant test statements

**Review by mchae-nava — APPROVED** [review]
Took a pass clicking through, LGTM! Minor considerations, more for future thought

- Minor comments/checks on tests
- If I remember, there was a lot of discussion on the ticket tracking this work, and some shifts on how we approached this. Might be good to include a light context in the PR description, as well as any steps planned in the near future following this work

File: spec/lib/disability_compensation/providers/lighthouse_rated_disabilities_provider_spec.rb (hunk 1 above)
Comment (mchae-nava) [reply spec/lib/disability_compensation/providers/lighthouse_rated_disabilities_provider_spec.rb:10]: I'm blind, I just noticed the includes, though perhaps the `shared_examples` could be made more robust and non-specific for even more reuse

**Review by ryan-mcneil — APPROVED** [review]
Thanks for the additional reviews! 🚀

**Pull request merged**